crates/factor-outbound-http/src/wasi.rs

```diff
@@ -104,6 +135,25 @@ impl WasiHttpView for WasiHttpImplInner<'_> {
         request: Request<wasmtime_wasi_http::body::HyperOutgoingBody>,
         config: wasmtime_wasi_http::types::OutgoingRequestConfig,
     ) -> wasmtime_wasi_http::HttpResult<wasmtime_wasi_http::types::HostFutureIncomingResponse> {
+        let connection_pooling = self.state.connection_pooling;
+        let builder = move || {
+            let mut builder = Client::builder(TokioExecutor::new());
+            if !connection_pooling {
+                builder.pool_max_idle_per_host(0);
+            }
+            builder
+        };
+
+        let http_clients = self
+            .state
+            .wasi_http_clients
+            .get_or_insert_with(|| HttpClients {
+                http1: builder().build(HttpConnector),
+                http2: builder().http2_only(true).build(HttpConnector),
+                https: builder().build(HttpsConnector),
```

Comment:
`Builder::build` takes `&self` (cloning internally) so you could reuse one `builder` here if you wanted. 🤷

Reply:
Yeah, but then the code would be order-dependent -- I'd have to be sure to build the `http2` one last so that the `http2_only` setting doesn't affect the other ones.  I'd prefer to contain the side effect by using separate builders unless there's a good reason not to.